packages/itwinui-react/src/core/Table/hooks/useColumnDragAndDrop.tsx

```diff
@@ -98,7 +98,7 @@ const defaultGetDragAndDropProps =
       event.preventDefault();
       setOnDragColumnStyle(event);
 
-      const columnIds = instance.flatHeaders.map((x) => x.id);
+      const columnIds = instance.allColumns.map((x) => x.id);
```

Comment:
Wasn't sure if it was better to use `allColumns` or `headers` here. `flatHeaders` was only returning the visible headers which was leading to the reordering issue on hidden columns

Reply:
I'm not sure either; if it's just getting the ids from them and those are the same it shouldn't matter thought.